src/v/raft/consensus.cc

```diff
@@ -2393,7 +2399,7 @@ ss::future<> consensus::hydrate_snapshot() {
     if (!metadata.has_value()) {
         co_return;
     }
-    update_offsets_from_snapshot(metadata.value());
+    auto commit_idx_updated = update_offsets_from_snapshot(metadata.value());
```

Comment:
i am wondering if it would be possible to do this as a last step in hydrate snapshot ? 
Currently it seems that the race still may happen. If a check in stm manager verifying if offset is ready to be applied is executed before the `co_await _configuration_manager.add` scheduling point, for that the notification is not required.